api/v1beta1/cinderscheduler_types.go

```diff
@@ -17,35 +17,67 @@ limitations under the License.
 package v1beta1
 
 import (
+	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // CinderSchedulerSpec defines the desired state of CinderScheduler
 type CinderSchedulerSpec struct {
-	// CR name of managing controller object to identify the config maps
-	ManagingCrName string `json:"managingCrName,omitempty"`
-	// Cinder Database Hostname String
-	DatabaseHostname string `json:"databaseHostname,omitempty"`
-	// Cinder Scheduler Container Image URL
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=cinder
+	// ServiceUser - optional username used for this service to register in cinder
+	ServiceUser string `json:"serviceUser"`
+
+	// +kubebuilder:validation:Optional
+	// ContainerImage - Cinder API Container Image URL
 	ContainerImage string `json:"containerImage,omitempty"`
-	// Cinder Scheduler Replicas
+
+	// +kubebuilder:validation:Required
+	// Replicas - Cinder API Replicas
 	Replicas int32 `json:"replicas"`
-	// Secret containing: CinderPassword, TransportURL
-	CinderSecret string `json:"cinderSecret,omitempty"`
-	// Secret containing: NovaPassword
-	NovaSecret string `json:"novaSecret,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// DatabaseHostname - Cinder Database Hostname
+	DatabaseHostname string `json:"databaseHostname,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=cinder
+	// DatabaseUser - optional username used for cinder DB, defaults to cinder
+	// TODO: -> implement needs work in mariadb-operator, right now only cinder
+	DatabaseUser string `json:"databaseUser,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Secret containing OpenStack password information for CinderDatabasePassword
+	Secret string `json:"secret,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// PasswordSelectors - Selectors to identify the DB and TransportURL from the Secret
+	PasswordSelectors PasswordSelector `json:"passwordSelectors,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// NodeSelector to target subset of worker nodes for running the Scheduler service
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Debug - enable debug for different deploy stages. If an init container is used, it runs and the
+	// actual action pod gets started with sleep infinity
+	Debug CinderDebug `json:"debug,omitempty"`
```

Comment:
should have customserviceconfig for scheduler specific overwrites? like cinder CRD has global cinder settings -> then each service has again customserviceconfig which wins over cinder crd customserviceconfig